### PR TITLE
Removed all warnings from VS compilation, and updated AlphaEncodingManager to use enum

### PR DESCRIPTION
**Pull request by noob247365**
A lot of minor tweaks like adding explicit casting, and one bigger tweak: Added an enum for storing the `'A'`/`'B'` values instead of hard-coding everywhere

### CODE REVIEW

File: include/Encode/AlphaEncodingManager.h [hunk 1]
```diff
@@ -2,6 +2,43 @@
...
+	(char)VRCommDataAlphaEncodingCharacter::JOY_BTN,
+	(char)VRCommDataAlphaEncodingCharacter::BTN_TRG,
+	(char)VRCommDataAlphaEncodingCharacter::BTN_A,
+	(char)VRCommDataAlphaEncodingCharacter::BTN_B,
+	(char)VRCommDataAlphaEncodingCharacter::GES_GRAB,
+	(char)VRCommDataAlphaEncodingCharacter::GES_PINCH,
+	(char)VRCommDataAlphaEncodingCharacter::BTN_MENU,
+	(char)VRCommDataAlphaEncodingCharacter::BTN_CALIB,
+	(char)0                                             // Turns into a null terminated string
+};
```
Comment (danwillm) [inline include/Encode/AlphaEncodingManager.h:40]: If these aren't used anywhere else, might be best to move them to the cpp file and make them static? I know they were defined in the header before, but now is probably a good time to move them

File: include/Encode/LegacyEncodingManager.h [hunk 2]
```diff
@@ -3,6 +3,23 @@
...
+  JOY_Y,
+  JOY_BTN,
+  BTN_TRG,
+  BTN_A,
+  BTN_B,
+  GES_GRAB,
+  GES_PINCH,
+  MAX,
+};
+
```
Comment (danwillm) [inline include/Encode/LegacyEncodingManager.h:22]: Likewise, I don't think this is used anywhere else

**Review by danwillm — CHANGES_REQUESTED** [review]
Thank you, these look like good changes. Just two comments

File: include/Encode/AlphaEncodingManager.h (hunk 1 above)
Comment (noob247365) [reply include/Encode/AlphaEncodingManager.h]: Fixed with latest commit

File: include/Encode/LegacyEncodingManager.h (hunk 2 above)
Comment (noob247365) [reply include/Encode/LegacyEncodingManager.h]: Fixed with latest commit

File: include/Encode/AlphaEncodingManager.h [hunk 3]
```diff
@@ -12,7 +49,7 @@ class AlphaEncodingManager : public IEncodingManager {
 
        private:
     std::string getArgumentSubstring(std::string str, char del);
+           bool argValid(std::string str, char del);
```
Comment (danwillm) [inline include/Encode/AlphaEncodingManager.h:15]: having `argValid` as a member isn't really necersarry,but making the function static in the cpp file would be a good idea.

**Review by danwillm — CHANGES_REQUESTED** [review]

File: include/Encode/AlphaEncodingManager.h (hunk 3 above)
Comment (noob247365) [reply include/Encode/AlphaEncodingManager.h]: Resolved with latest commit

**Review by danwillm — APPROVED** [review]

**Pull request merged**